block-producer/src/block_builder/prover/block_witness.rs

```diff
@@ -33,6 +33,8 @@ pub struct BlockWitness {
     pub(super) updated_accounts: BTreeMap<AccountId, AccountUpdate>,
     /// (batch_index, created_notes_root) for batches that contain notes
     pub(super) batch_created_notes_roots: BTreeMap<usize, Digest>,
+    pub(super) produced_nullifiers: BTreeMap<Digest, MerklePath>,
+    pub(super) nullifier_smt_leaves: Vec<SmtLeaf>,
```

Comment:
Would it make sense to consolidate these into a single field? Since `StmProof` is basically leaf + path, maybe something like this would work:
```Rust
pub(super) produced_nullifiers: BTreeMap<Digest, SmtProof>,
```

Reply:
I had this initially, but then we would need to add an accessor `SmtProof::get_leaf() -> &SmtLeaf` (i.e. splitting in 2 variables is a result of only having `into_parts(self)`).

Should I open a PR on `miden-crypto` to address that?

Reply:
Refactored!